### PR TITLE
[Lim Boon Hai] iP

### PR DESCRIPTION
**Pull request by boonhaii**
# Duii Bot
> "Your mind is for having ideas, not for holding them" - David Allen [[Source](https://dansilvestre.com/productivity-quotes)]

Duii Bot is a bot which helps you track your tasks and its:
* Text-based
* Simple to use
* Coded in Java!

To start using the bot:
1. Download the program [here](https://github.com/boonhaii/ip/releases/tag/v0.1).
2. Open up terminal, and locate the newly downloaded file on your computer.
3. Run the file by typing `java -jar ./ip.jar`.
4. Follow the instructions given by the bot and watch it keep track of your tasks for you!

You would be doing us a *great* favour by using this bot and supporting this *free* project.

Current support features:
- [x] Tracking of tasks
- [x] Tracking of timed tasks, such as Deadlines and Events
- [ ] Reminders by the bot (coming soon!) 

Oh and if you are a Java Programmer yourself, you can take this opportunity to showcase your Java skills!
Create a new Java file and write the code below before running.
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Thank you for supporting our project!

### CODE REVIEW

File: src/main/java/duke/UI.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Task;
+
+/**
+ * UI class which handles all the interactions with the user.
+ */
+public class UI {
```
Comment (LMAOboxhack) [inline src/main/java/duke/UI.java]:
```suggestion
public class Ui {
```

Should your class name be in lowercase instead? The [Naming](https://se-education.org/guides/conventions/java/intermediate.html#naming) conventions of the module coding standard state that "abbreviations and acronyms should not be uppercase when used as a part of a name".

File: src/main/java/duke/Storage.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+     * @throws DukeException If an IOException occurs upon reading of file, which is passed on to
+     *                       the main() method of Duke class.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        try {
+            file = this.getfiles();
+            FileReader reader = new FileReader(file);
+            BufferedReader buffreader = new BufferedReader(reader);
+            String text;
+            ArrayList<Task> saved = new ArrayList<>();
```
Comment (LMAOboxhack) [inline src/main/java/duke/Storage.java:83]:
```suggestion
            String text;
            
            ArrayList<Task> saved = new ArrayList<>();
```

Might be good to add a blank line between your variable initializations and your loading logic? The [Layout](https://se-education.org/guides/conventions/java/intermediate.html#layout) section of the course coding standard states that "logical units within a block should be separated by one blank line".

File: src/main/java/duke/Storage.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        }
+    }
+
+    /**
+     * Writes the tasks of the current session into the dynamically created file
+     * on the local filesystem.
+     *
+     * @throws DukeException If an IOException occurs during writing of file, which
+     *                       is passed on to the main() method of Duke class.
+     */
```
Comment (LMAOboxhack) [inline src/main/java/duke/Storage.java]:
```suggestion
    /**
     * Writes the tasks of the current session into the dynamically created file
     * on the local filesystem.
     * 
     * @param list ...
     *
     * @throws DukeException If an IOException occurs during writing of file, which
     *                       is passed on to the main() method of Duke class.
     */
```

Would it be a good idea to add Javadoc comments for method parameters as well? Might help fellow developers understand what the inputs to your methods are. The form of Javadoc comments is specified in the [Comments](https://se-education.org/guides/conventions/java/intermediate.html#comments) section of the course coding standard.

File: src/main/java/duke/UI.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+    private static String logo = " ____        _ _\n"
```
Comment (LMAOboxhack) [inline src/main/java/duke/UI.java]:
```suggestion
    private static String LOGO = " ____        _ _\n"
```
Would it be good for this constant name to be capitalized? This is mentioned in the [Naming](https://se-education.org/guides/conventions/java/intermediate.html#naming) conventions of the module coding standard.

File: src/main/java/duke/Storage.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    /**
+     * Searches the current file system for existing saved files.
+     * Creates a new data directory consisting of the text file to write
+     * into if saved file does not exist.
+     *
+     * @return The abstract representation of the pathname of the created file
+     */
+    public File getfiles() {
```
Comment (LMAOboxhack) [inline src/main/java/duke/Storage.java]:
```suggestion
    public File getFiles() {
```

Would it be good to change the method name to follow camelCase convention? This is outlined in the module's [Naming](https://se-education.org/guides/conventions/java/intermediate.html#naming) conventions.

**Review by LMAOboxhack — COMMENTED** [review]
Looks good, just a few minor nits! 👍

**Comment by boonhaii** [comment]
> Looks good, just a few minor nits! 👍

Thanks for the review!

File: src/main/java/duke/Duke.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    private Storage storage;
+    private Ui userInt;
+    private TaskList tasks;
+    private User user;
+
+    /**
+     * Constructor of Duke instance, which initalises the other
+     * components of the project.
+     *
+     * @param user User object which encapsulates the users' name and visit history
```
Comment (Ongzl) [inline src/main/java/duke/Duke.java]: think you missed a fullstop here

File: src/main/java/duke/Duke.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            userInt.showLoadingError();
+            this.tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Gets the string representation of the response by the User Interface.
+     *
+     * @param input The input string by the user
+     * @return The string to be printed out to the GUI as the response
```
Comment (Ongzl) [inline src/main/java/duke/Duke.java:33]: Missing fullstop here too

File: src/main/java/duke/Storage.java [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    public Storage() {
+        this.dateTimeFormat = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm");
+    }
+
+    /**
+     * Searches the current file system for existing saved files.
+     * Creates a new data directory consisting of the text file to write
+     * into if saved file does not exist.
+     *
+     * @return The abstract representation of the pathname of the created file
```
Comment (Ongzl) [inline src/main/java/duke/Storage.java:39]: Missing fullstop here too

File: src/main/java/duke/command/AddCommand.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * AddCommand class which indicates input is adding new tasks.
+ */
+public class AddCommand extends Command {
+    private boolean isExit = false;
+    private Task newTask;
+
+    /**
+     * Public constructor for the AddCommand.
+     * @param newTask The task to be added.
```
Comment (Ongzl) [inline src/main/java/duke/command/AddCommand.java:22]: Think you're missing a line break here

File: src/main/java/duke/command/DeleteCommand.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+
+public class DeleteCommand extends Command {
+    private boolean isExit = false;
+    private int taskid;
+    private Task removedTask;
+
+    /**
+     * Public constructor of the DeleteCommand object.
+     * @param taskid The id of the task to be removed, with respect to
```
Comment (Ongzl) [inline src/main/java/duke/command/DeleteCommand.java]: Missing line break here too

File: src/main/java/duke/command/DoneCommand.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+import duke.task.Task;
+
+public class DoneCommand extends Command {
+    private boolean isExit = false;
+    private int taskid;
+    private Task doneTask;
+
+    /**
+     * Public constructor of the DoneCommand object.
+     * @param taskid The id of the task to be removed, with respect to
```
Comment (Ongzl) [inline src/main/java/duke/command/DoneCommand.java]: Missing line break here too

File: src/main/java/duke/command/FindCommand.java [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+import java.util.ArrayList;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+
+/**
+ * FindCommand class which searches for tasks with matching keywords
```
Comment (Ongzl) [inline src/main/java/duke/command/FindCommand.java]: Missing full stop

File: src/main/java/duke/command/FindCommand.java [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+     *
+     * @param keyword The keyword used to search for the matching tasks.
+     */
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    /**
+     * Lists the tasks with the matching keyword associated with the FindCommand.
+     *  @param tasks The TaskList associated with the current bot.
```
Comment (Ongzl) [inline src/main/java/duke/command/FindCommand.java]: Missing line break

File: src/main/java/duke/task/Event.java [hunk 14]
```diff
@@ -0,0 +1,59 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Tasks which occur during a certain time
```
Comment (Ongzl) [inline src/main/java/duke/task/Event.java:7]: Missing full stop

File: src/main/java/duke/Parser.java [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+
+public class Parser {
+    /**
+     * Parses the input from the user.
+     *
+     * @param input The input inserted by the user.
+     * @return The respective commands to be executed based on the input.
+     * @throws DukeException This exception is thrown when the input is invalid.
+     */
+    public static Command parse(String input) throws DukeException {
```
Comment (Ongzl) [inline src/main/java/duke/Parser.java:38]: This method / while loop might be too long. This [coding quality website](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)  stated the following:

>Be wary when a method is longer than the computer screen, and take corrective action when it goes
>beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.

This is a little longer than 30 LOC , though I too am not sure how to condense this part of logic too haha

**Review by Ongzl — COMMENTED** [review]
LGTM except some javadocs missing full stops and line breaks, cheers

**Pull request closed without merging**